projects/material-addons/src/lib/table-filter-overlay/table-filter-overlay.component.ts

```diff
@@ -0,0 +1,175 @@
+    filter(o => !o),
+    map(() => {}),
+  );
+
+  positions: ConnectedPosition[] = [
+    {
+      originX: 'start',
+      originY: 'top',
+      overlayX: 'start',
+      overlayY: 'top',
+    },
+    {
+      originX: 'start',
+      originY: 'bottom',
+      overlayX: 'start',
+      overlayY: 'bottom',
+    },
+  ];
+
+  transformOrigin = 'top';
+  triggerFontSize = 0;
+  offsetY = 0;
+  panelDoneAnimatingStream = new Subject<string>();
+  triggerRect: ClientRect;
+  scrollStrategy: ScrollStrategy;
+
+  @ViewChild(CdkConnectedOverlay, { static: false }) overlayDir: CdkConnectedOverlay;
+  @ViewChild('trigger', { static: false }) trigger: ElementRef;
+  @ViewChild('panel', { static: false }) panel: ElementRef;
+
+  private isOpen: boolean;
+  private readonly scrollStrategyFactory: () => ScrollStrategy;
+  private readonly destroy$ = new Subject<void>();
+
+  get panelOpen() {
+    return this.isOpen;
+  }
+
+  constructor(
+    private overlay: Overlay,
+    private cdr: ChangeDetectorRef,
+    private viewportRuler: ViewportRuler,
+    @Optional() private parentFormField: MatFormField,
+    @Inject(MAT_SELECT_SCROLL_STRATEGY) scrollStrategyFactory: any,
+  ) {
+    this.scrollStrategyFactory = scrollStrategyFactory;
+    this.scrollStrategy = this.scrollStrategyFactory();
+  }
+
+  ngOnDestroy() {
+    this.destroy$.next();
+    this.destroy$.complete();
+  }
+
+  ngOnInit(): void {
+    this.panelDoneAnimatingStream.pipe(distinctUntilChanged(), takeUntil(this.destroy$)).subscribe(() => {
+      if (this.panelOpen) {
+        this.openedChange.emit(true);
+      } else {
+        this.openedChange.emit(false);
+        this.overlayDir.offsetX = 0;
+        this.cdr.markForCheck();
+      }
+    });
+
+    this.viewportRuler
+      .change()
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(() => {
+        if (this.isOpen) {
+          this.triggerRect = this.trigger.nativeElement.getBoundingClientRect();
+          this.cdr.markForCheck();
+        }
+      });
+  }
+
+  close(): void {
+    if (this.isOpen) {
+      this.isOpen = false;
+      this.cdr.markForCheck();
+    }
+  }
+
+  open(): void {
+    if (this.disabled || this.isOpen) {
+      return;
+    }
+
+    this.triggerRect = this.trigger.nativeElement.getBoundingClientRect();
+    this.triggerFontSize = parseInt(getComputedStyle(this.trigger.nativeElement).fontSize || '0', 10);
+    this.isOpen = true;
+    this.cdr.markForCheck();
+  }
+
+  toggle(): void {
+    if (this.panelOpen) {
+      this.close();
+    } else {
+      this.open();
+    }
+  }
+
+  onAttached(): void {
+    this.overlayDir.positionChange.pipe(take(1)).subscribe(() => {
+      this.cdr.detectChanges();
+      this.calculateOverlayOffsetX();
+    });
+  }
+
+  getPanelTheme(): string {
+    return this.parentFormField ? `mat-${this.parentFormField.color}` : '';
+  }
+
+  private calculateOverlayOffsetX(): void {
```

Comment:
Do not understand everything that is done in this component - we possibly need to talk in person